source/agora/consensus/PreImage.d

```diff
@@ -328,70 +318,6 @@ public struct PreImageCycle
 
     public PreImageCache preimages;
 
-    /***************************************************************************
-
-        Populate the caches of pre-images
-
-        This will first populate the caches (seeds and preimages) as necessary,
-        then increase the `index` by one, or reset it to 0 and increase `nonce`
-        if necessary.
-
-        Note that the increment is done after the caches are populated,
-        so `populate` needs to be called once this node has confirmed it
-        is part of consensus. If the caches are only needed without the
-        increment, the `consume` parameter must be set to `false`.
-
-        Params:
-          secret = The secret key of the node, used as part of the hash
-                   to generate the cycle seeds
-          consume = If true, calculate the cycle index. Otherwise, just
-                   get the random seed which is the first pre-image
-
-        Returns:
-          The hash of the current enrollment round
-
-    ***************************************************************************/
-
-    public Hash populate (scope const ref Scalar secret, bool consume)
-        @safe nothrow
-    {
-        // Populate the nonce cache if necessary
-        if (this.index == 0)
-        {
-            const cycle_seed = hashMulti(
-                secret, "consensus.preimages", this.nonce);
-            this.seeds.reset(cycle_seed);
-        }
-
-        if (consume)
-        {
-            // Populate the current enrollment round cache
-            // The index into `seeds` is the absolute index of the cycle,
-            // not the number of round, hence why we use `byStrides`
-            // The alternative would be:
-            // [$ - (this.index + 1) * this.preimages.length]
-            this.preimages.reset(this.seeds.byStride[$ - 1 - this.index]);
-
-            // Increment index if there are rounds left in this cycle
-            this.index += 1;
-            if (this.index >= NumberOfCycles)
-            {
-                this.index = 0;
-                this.nonce += 1;
-            }
-            return this.preimages[$ - 1];
-        }
-        else
-        {
-            // If this is used for getting initial pre-image for enrollment
-            // it just creates initial pre-image of the cycle seed.
-            auto next_images = PreImageCache(this.preimages.data.length,
-            this.preimages.interval);
-            next_images.reset(this.seeds.byStride[$ - 1 - this.index]);
-            return next_images[$ - 1];
-        }
-    }
-
     /***************************************************************************
 
         Seek to the `PreImage`s at height `height`
```

Comment:
The documentation still refers to `nonce`.

Reply:
Updated